build.gradle.kts

```diff
@@ -134,90 +140,113 @@ tasks.withType<JacocoReport> {
+    println("See: https://github.com/javafxports/openjdk-jfx/issues/66")
+    // Link: https://github.com/javafxports/openjdk-jfx/issues/66
+    useJUnitPlatform {
+        excludeTags("UI")
     }
 }
 
-/*
-     * Allows you to run the UI tests in headless mode by calling gradle with the -Pheadless argument
-     */
-if (project.hasProperty("jenkinsBuild") || project.hasProperty("headless")) {
-    println("Running UI Tests Headless")
-    junitPlatform {
-        filters {
-            tags {
-                /*
-                 * A category for UI tests that cannot run in headless mode, ie work properly with real windows
-                 * but not with the virtualized ones in headless mode.
-                 */
-                exclude("NonHeadlessTests")
-            }
-        }
+tasks.withType<Javadoc> {
+    isFailOnError = false
+}
+
+dependencies {
+}
+
+val nativeShadowTasks = NativePlatforms.values().map { platform ->
+    tasks.create<ShadowJar>("shadowJar-${platform.platformName}") {
+        classifier = platform.platformName
+        configurations = listOf(
+                project.configurations.compile,
+                project.configurations.getByName(platform.platformName)
+        )
+        from(
+                java.sourceSets["main"].output
+        )
     }
-    tasks {
-        "junitPlatformTest"(JavaExec::class) {
-            jvmArgs = listOf(
-                    "-Djava.awt.headless=true",
-                    "-Dtestfx.robot=glass",
-                    "-Dtestfx.headless=true",
-                    "-Dprism.order=sw",
-                    "-Dprism.text=t2k"
-            )
-        }
+}
+
+tasks.create("shadowJarAllPlatforms") {
+    nativeShadowTasks.forEach {
+        this.dependsOn(it)
     }
 }
 
+tasks.withType<ShadowJar>().configureEach {
+    exclude("module-info.class")
+}
+
+val sourceJar = task<Jar>("sourceJar") {
+    description = "Creates a JAR that contains the source code."
+    from(java.sourceSets["main"].allSource)
+    classifier = "sources"
+}
+
+val javadocJar = task<Jar>("javadocJar") {
+    dependsOn("javadoc")
+    description = "Creates a JAR that contains the javadocs."
+    from(java.docsDir)
+    classifier = "javadoc"
+}
+
 publishing {
     publications {
-        create<MavenPublication>("OutlineViewer") {
-            artifactId = "OutlineViewer"
-            getWPILibVersion()?.let { version = it }
-            val shadowJar: ShadowJar by tasks
-            artifact (shadowJar) {
-                classifier = null
+        create<MavenPublication>("app") {
+            groupId = "edu.wpi.first.outlineviewer"
+            artifactId = "outlineviewer"
+            version = project.version as String
+            nativeShadowTasks.forEach {
+                artifact(it) {
+                    classifier = it.classifier
+                }
             }
+            artifact(sourceJar)
+            artifact(javadocJar)
```

Comment:
I don't think we need to bother publishing sources and javadocs for a standalone application like OutlineViewer